pkgs/stdenv/booter.nix

```diff
@@ -57,12 +57,18 @@ stageFuns: let
   # debugging purposes.
   folder = stageFun: finalSoFar: let
     args = stageFun finalSoFar;
-    stdenv = args.stdenv // {
-      # For debugging
-      __bootPackages = finalSoFar;
+    args' = args // {
+      stdenv = args.stdenv // {
+        # For debugging
+        __bootPackages = finalSoFar;
+      };
     };
-    args' = args // { inherit stdenv; };
-  in
-    (if args.__raw or false then lib.id else allPackages) args';
+    self =
+      if args.__raw or false
```

Comment:
In practice `__raw` is early ad-hoc `rec { .. }`. They usually just `import` rather than `callPackage`. Going forward I wonder if we can provide more structure to those bootstrapping stages, for now this is the quickest way to keep things working.